### ISSUE TITLE
case-insensitive import collision

### ISSUE THREAD

**Issue by tachang** [FairwindsOps/rbac-lookup#43]
Seem to be getting this error when running make.

can't load package: package github.com/FairwindsOps/rbac-lookup/cmd: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/cmd" and "github.com/fairwindsops/rbac-lookup/cmd"

**Comment by sudermanjr** [comment]
Hi, thanks for filing the issue.

I'm currently looking into it, but could you please share your Golang version, your OS, and whether or not you have the repo cloned as `FairwindsOps` or `fairwindsops`?

**Comment by tachang** [comment]
Sure. That said `make build` works fine.

I did a `git clone git@github.com:FairwindsOps/rbac-lookup.git` which resulted in a `rbac-lookup` directory.

I am on `Ubuntu 18.04.2 LTS`.


```
$ go version
go version go1.12 linux/amd64
```

**Comment by sudermanjr** [comment]
> Sure. That said `make build` works fine.

I'm a little confused. Are you saying that `make` doesn't work, but `make build` does work?

What does `make test` do?

**Comment by tachang** [comment]
That's correct. It looks like `make test` is the issue.

```
$ make test
printf "Linter:\n"
Linter:
GO111MODULE=on GO111MODULE=on go list ./... | xargs -L1 golint | tee golint-report.out
can't load package: package github.com/FairwindsOps/rbac-lookup/cmd: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/cmd" and "github.com/fairwindsops/rbac-lookup/cmd"
can't load package: package github.com/FairwindsOps/rbac-lookup/lookup: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/lookup" and "github.com/fairwindsops/rbac-lookup/lookup"
xargs: golint: No such file or directory
printf "\n\nTests:\n\n"


Tests:

GO111MODULE=on GO111MODULE=on go test -v --bench --benchmem -coverprofile coverage.txt -covermode=atomic ./...
can't load package: package github.com/FairwindsOps/rbac-lookup/cmd: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/cmd" and "github.com/fairwindsops/rbac-lookup/cmd"
can't load package: package github.com/FairwindsOps/rbac-lookup/lookup: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/lookup" and "github.com/fairwindsops/rbac-lookup/lookup"
Makefile:14: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by tachang** [comment]
Oh I don't have golint.

**Comment by tachang** [comment]
Just did
`$ go get -u golang.org/x/lint/golint`

```
$ make test
printf "Linter:\n"
Linter:
GO111MODULE=on GO111MODULE=on go list ./... | xargs -L1 golint | tee golint-report.out
go: downloading golang.org/x/net v0.0.0-20200114155413-6afb5195e5aa
go: downloading golang.org/x/crypto v0.0.0-20200117160349-530e935923ad
go: downloading golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
go: downloading golang.org/x/sys v0.0.0-20200122134326-e047566fdf82
go: extracting golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
go: extracting golang.org/x/crypto v0.0.0-20200117160349-530e935923ad
go: extracting golang.org/x/net v0.0.0-20200114155413-6afb5195e5aa
go: downloading golang.org/x/text v0.3.2
go: extracting golang.org/x/sys v0.0.0-20200122134326-e047566fdf82
go: extracting golang.org/x/text v0.3.2
can't load package: package github.com/FairwindsOps/rbac-lookup/cmd: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/cmd" and "github.com/fairwindsops/rbac-lookup/cmd"
can't load package: package github.com/FairwindsOps/rbac-lookup/lookup: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/lookup" and "github.com/fairwindsops/rbac-lookup/lookup"
xargs: golint: No such file or directory
printf "\n\nTests:\n\n"


Tests:

GO111MODULE=on GO111MODULE=on go test -v --bench --benchmem -coverprofile coverage.txt -covermode=atomic ./...
can't load package: package github.com/FairwindsOps/rbac-lookup/cmd: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/cmd" and "github.com/fairwindsops/rbac-lookup/cmd"
can't load package: package github.com/FairwindsOps/rbac-lookup/lookup: case-insensitive import collision: "github.com/FairwindsOps/rbac-lookup/lookup" and "github.com/fairwindsops/rbac-lookup/lookup"
Makefile:14: recipe for target 'test' failed
make: *** [test] Error 1
```

**Comment by sudermanjr** [comment]
Thanks for the update. We will look into it further.

**Comment by sudermanjr** [comment]
@tachang I noticed that the go.mod file was not checked in. Did you generate it using `go mod init` before you ran `make`?

The makefile automatically enables modules, which could have caused issues. Could you try #44 and let me know if that helps?

**Comment by tachang** [comment]
Didn't run anything before. I'm fairly new to golang so I just ran make and hoped it worked :)

After you didn't I looked into the Makefile just to see if there were any other targets. I did make build and that worked so I figured I didn't need the test part.

**Comment by sudermanjr** [comment]
> Didn't run anything before. I'm fairly new to golang so I just ran make and hoped it worked :)
> 
> After you didn't I looked into the Makefile just to see if there were any other targets. I did make build and that worked so I figured I didn't need the test part.

Gotcha. I'm a little surprised that it worked at all, considering that master did not have a go.mod and a go.sum file. If I delete those files and run make, I get an error like this (which is expected)

```
└─ make
printf "Linter:\n"
Linter:
GO111MODULE=on go list ./... | xargs -L1 golint | tee golint-report.out
go: cannot find main module, but found .git/config in /Users/asuderma/go/src/github.com/fairwindsops/rbac-lookup
        to create a module there, run:
        go mod init
printf "\n\nTests:\n\n"


Tests:

GO111MODULE=on go test -v --bench --benchmem -coverprofile coverage.txt -covermode=atomic ./...
go: cannot find main module, but found .git/config in /Users/asuderma/go/src/github.com/fairwindsops/rbac-lookup
        to create a module there, run:
        go mod init
make: *** [test] Error 1
```

I merged #44, so if you could delete your clone of rbac-lookup and then try cloning again and running make on master I would love to hear the results. 

Thanks!

**Comment by sudermanjr** [comment]
This does not seem to be repeatable. Please feel free to re-open if you have more information.
